Input/ClickHandler.gd

```diff
@@ -49,5 +61,25 @@ func _on_mouse_entered() -> void:
 func _on_mouse_exited() -> void:
 	if not _is_interactable:
 		return
-		
+	
+	_set_lock_hover_timer()
+	
 	on_unhover.emit()
+
+
+func _create_lock_hover_timer():
+	_lock_hover_timer = Timer.new()
+	add_child(_lock_hover_timer)
+	
+	# When timer expires, enable hovering
+	_lock_hover_timer.timeout.connect(func(): _hover_enabled = true)
```

Comment:
this lambda function is created withing the scope of `_create_lock_hover_time`
even if you connect it, does it stay after you exit the function ? maybe this should be declared as a function in itself ?

Reply:
Yes the lambda stays from my testing. If we connect in the function itself it's going to connect each time that card is unhovered.

Reply:
can we do something to connect the function only once maybe ? using `is_connected`
my problem is that I fail to understand how the function stays even though it's declared within the scope of another function. I feel like it should just disappear once the function has finished running
Maybe the fact that it is connected to something keeps it ?

Reply:
I don't really know how lambda's work under the hood tbh, so I can't say I understand how it stays connected either. If you're worried about it, I can just move that lambda out into a separate function, like
```GDScript
_lock_hover_timer.timeout.connect(_on_lock_hover_timeout)

func _on_lock_hover_timeout():
   _hover_enabled = true
```

Reply:
If it works with the lambda function properly, it could be nice to understand why :thinking:

Reply:
The answer if some people were to look at this discussion later
![image](https://github.com/Saplings-Projects/1M_sub/assets/49365680/6fa10802-6e4e-4315-8118-0553ef8d46e8)